.github/workflows/issue-write.yml

```diff
@@ -0,0 +1,89 @@
+name: Comment on an issue
+
+on:
+  workflow_run:
+    workflows: ["Check code formatting"]
+    types:
+      - completed
+
+permissions:
+  contents: read
+
+jobs:
+  pr-comment:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+    if: >
+      github.event.workflow_run.event == 'pull_request'
+    steps:
+      - name: 'Download artifact'
```

Comment:
I think i you use the newer [upload-artifacts](https://github.com/actions/upload-artifact) and [download-artifacts](https://github.com/actions/download-artifact) packages, then you don't need to deal with all the listing/downloading/unzip code -- it'll do it for you just based on a name and run_id.